script/add-grammar

```diff
@@ -43,24 +43,32 @@ def log(msg)
   puts msg if $verbose
 end
 
+$aborted = false
+
 def command(*args)
   log "$ #{args.join(' ')}"
   output, status = Open3.capture2e(*args)
   if !status.success?
-    output.each_line do |line|
-      log "  > #{line}"
+    output = output.each_line { |line| "  > #{line}" }
+    unless $aborted
+      $aborted = true
+      warn "Command failed. Aborting."
+      raise output
+    else
+      warn output
```

Comment:
This seems overly complicated on quick inspection. 

Why do we need to differentiate behaviour here? Wouldn't we want to _always_ exit and clean up on failure?

Reply:
If the command raises an exception, the output is discarded entirely. The globals appear scoped to the containing block, as they weren't showing up when output in the `rescue` part.